### PR TITLE
Validation for percentage amt for Cart and Catalog Rule.

### PR DESCRIPTION
**Pull request by papnoisanjeev**
## Issue Reference
#6554

## Description
Added validation for percentage amount for Catalog and Cart Rule and Handled validation using **FormRequest** and removed validation from controller.

### CODE REVIEW

File: packages/Webkul/CartRule/src/Http/Controllers/CartRuleController.php [hunk 1]
```diff
@@ -94,23 +95,10 @@ public function copy(int $cartRuleId): View
      *
```
Comment (devansh-webkul) [inline packages/Webkul/CartRule/src/Http/Controllers/CartRuleController.php:95]: Add PHPDoc also.

File: packages/Webkul/CartRule/src/Http/Controllers/CartRuleController.php [hunk 2]
```diff
@@ -146,21 +134,9 @@ public function edit($id)
      * @param  int  $id
      * @return \Illuminate\Http\Response
      */
-    public function update(Request $request, $id)
+    public function update(CartRuleRequest $cartRuleRequest, $id)
```
Comment (devansh-webkul) [inline packages/Webkul/CartRule/src/Http/Controllers/CartRuleController.php:137]: Same here.

File: packages/Webkul/CatalogRule/src/Http/Controllers/CatalogRuleController.php [hunk 3]
```diff
@@ -60,19 +61,9 @@ public function create()
      *
```
Comment (devansh-webkul) [inline packages/Webkul/CatalogRule/src/Http/Controllers/CatalogRuleController.php:61]: Same here.

File: packages/Webkul/CatalogRule/src/Http/Controllers/CatalogRuleController.php [hunk 4]
```diff
@@ -103,21 +94,11 @@ public function edit($id)
      * @param  int  $id
```
Comment (devansh-webkul) [inline packages/Webkul/CatalogRule/src/Http/Controllers/CatalogRuleController.php:94]: Same here.

**Review by devansh-webkul — CHANGES_REQUESTED** [review]

File: packages/Webkul/CatalogRule/src/Http/Controllers/CatalogRuleController.php (hunk 4 above)
Comment (papnoisanjeev) [reply packages/Webkul/CatalogRule/src/Http/Controllers/CatalogRuleController.php:95]: Done.

**Review by devansh-webkul — APPROVED** [review]

**Comment by Sanjaybhattwebkul** [comment]
@papnoisanjeev 
The page should be auto focus on error

**Comment by papnoisanjeev** [comment]
@Sanjaybhattwebkul 

As validation is from backend so most probably auto focus will not work here.

**Review by Sanjaybhattwebkul — APPROVED** [review]

**Pull request merged**